modules/EvseManager/BspStateMachine.cpp

```diff
@@ -0,0 +1,362 @@
+            // within 100ms
+            // EVLOG_info << "statemachine allow_power_on false";
+            power_on_allowed = false;
+            power_on_reason = types::evse_board_support::Reason::NormalStop;
+            internal_allow_power_on(false);
+        }
+
+        ventilated_charging_active = false;
+
+        // Table A.6: Sequence 1.1 Plug-in
+        if (last_cp_state == RawCPState::A || last_cp_state == RawCPState::Disabled) {
+            event(CPEvent::CarPluggedIn);
+            ev_simplified_mode = false;
+        }
+
+        if (last_cp_state == RawCPState::E || last_cp_state == RawCPState::F) {
+            event(CPEvent::EFtoBCD);
+        }
+        break;
+
+    case RawCPState::C:
+        // Table A.6: Sequence 1.2 Plug-in
+        if (last_cp_state == RawCPState::A) {
+            event(CPEvent::CarPluggedIn);
+            EVLOG_info << "Detected simplified mode.";
+            ev_simplified_mode = true;
+        }
+        if (last_cp_state == RawCPState::B) {
+            event(CPEvent::CarRequestedPower);
+            if (power_on_allowed) {
+                internal_allow_power_on(true);
+            }
+        }
+
+        if (!pwm_running && last_pwm_running) { // X2->C1
+                                                // Table A.6 Sequence 10.2: EV does not stop drawing power
+                                                // even if PWM stops. Stop within 6 seconds (E.g. Kona1!)
+            timeout_state_c1.start(std::chrono::milliseconds(6000));
+        }
+
+        if (timeout_state_c1.reached()) {
+            // We are still in state C, but the 6 seconds timeout has been reached. Now force power off under load.
+            power_on_reason = types::evse_board_support::Reason::NormalStop;
+            internal_allow_power_on(false);
+        }
+
+        if (pwm_running) { // C2
+            // delayed power on
+            if (power_on_allowed && !last_power_on_allowed) {
+                // Table A.6: Sequence 4 EV ready to charge.
+                // Must enable power within 3 seconds.
+                internal_allow_power_on(true);
+            }
+
+            // Simulate Request power Event here for simplified mode
+            // to ensure that this mode behaves similar for higher
+            // layers. Note this does not work with 5% mode
+            // correctly, but simplified mode does not support HLC
+            // anyway.
+            if (!last_pwm_running && ev_simplified_mode) {
+                event(CPEvent::CarRequestedPower);
+            }
+        }
+
+        break;
+
+    case RawCPState::D:
+        // not supported
+        event(CPEvent::ErrorVentilationNotAvailable);
+        break;
+
+    case RawCPState::E:
+        if (last_cp_state != RawCPState::E) {
+            event(CPEvent::ErrorE);
+        }
+        break;
+
+    case RawCPState::F:
+        // this can only be set by evse, so we do not need to act here
```

Comment:
timeout stop added. F is handled by Charger fully, but I added power off here as well because it does not hurt. Locking/unlocking should be handled by the BSP, ideally on the MCU where the lowest level BSP code is running (you still want to unlock even if linux crashes)